sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/BulkWriter.scala

```diff
@@ -130,6 +143,219 @@ class BulkWriter(container: CosmosAsyncContainer,
+                          // for easier exception and semaphore handling
+                          // However a consequences of it could be, the generated bulk item operation will fail due to conflicts or pre-condition failure
+                          // If this turns out to be a problem, we can do more optimization here: merge multiple cosmosPatchBulkUpdateOperations into one bulkItemOperation
+                          // But even the above approach can only work within the same batch but not for the whole spark partition processing.
+                          for (readManyOperation <- readManyOperations) {
+                              val cosmosPatchBulkUpdateOperations =
+                                  cosmosPatchHelperOpt
+                                      .get
+                                      .createCosmosPatchBulkUpdateOperations(readManyOperation.objectNode)
+
+                              val rootNode =
+                                  cosmosPatchHelperOpt
+                                      .get
+                                      .patchBulkUpdateItem(resultMap.get(readManyOperation.cosmosItemIdentity), cosmosPatchBulkUpdateOperations)
+
+                              // create bulk item operation
+                              val etagOpt = Option.apply(rootNode.get(CosmosConstants.Properties.ETag))
+                              val bulkItemOperation = etagOpt match {
+                                  case Some(etag) =>
+                                      CosmosBulkOperations.getReplaceItemOperation(
+                                          readManyOperation.cosmosItemIdentity.getId,
+                                          rootNode,
+                                          readManyOperation.cosmosItemIdentity.getPartitionKey,
+                                          new CosmosBulkItemRequestOptions().setIfMatchETag(etag.asText()),
+                                          OperationContext(
+                                              readManyOperation.operationContext.itemId,
+                                              readManyOperation.operationContext.partitionKeyValue,
+                                              Some(etag.asText()),
+                                              readManyOperation.operationContext.attemptNumber,
+                                              Some(readManyOperation.objectNode)
+                                          ))
+                                  case None => CosmosBulkOperations.getCreateItemOperation(
+                                      rootNode,
+                                      readManyOperation.cosmosItemIdentity.getPartitionKey,
+                                      OperationContext(
+                                          readManyOperation.operationContext.itemId,
+                                          readManyOperation.operationContext.partitionKeyValue,
+                                          None,
```

Comment:
NIT: variable name like in c# is possible in scala and would help readability here